mirrord/operator/src/crd/mysql_branching.rs

```diff
@@ -0,0 +1,74 @@
+use chrono::{DateTime, Utc};
+use k8s_openapi::api::core::v1::ResourceRequirements;
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(CustomResource, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[kube(
+    group = "dbs.mirrord.metalbear.co",
+    version = "v1alpha1",
+    kind = "MysqlBranchDatabase",
+    status = "MysqlBranchDatabaseStatus",
+    namespaced
+)]
+pub struct MysqlBranchDatabaseSpec {
+    /// Identifier of this branch DB.
+    pub id: String,
+    /// Database connection info from the workload.
+    pub connection_source: ConnectionSource,
+    /// The duration in seconds this branch database will live idling.
+    pub ttl_secs: u64,
+    /// MySQL server image.
+    pub image: Option<String>,
+    /// Database pod resource requirements.
+    pub resource_requirements: Option<ResourceRequirements>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub enum ConnectionSource {
+    /// A complete connection URL.
+    Url(ConnectionSourceKind),
+    /// A group of connection parameters.
+    Parameters {
+        host: ConnectionSourceKind,
+        port: Option<ConnectionSourceKind>,
+        username: ConnectionSourceKind,
+        password: ConnectionSourceKind,
+        database: ConnectionSourceKind,
+    },
+}
```

Comment:
I think we also want to support the user specifying a `usernameSource`, a `passwordSource` and a `connectionEnvVar` as per the issue, where the connection url does not contain credentials.